app/controllers/work_packages/shares_controller.rb

```diff
@@ -40,25 +40,32 @@ def index
   end
 
   def create
-    user_id = params[:member][:user_id]
-
-    # In case, the user does not exist yet: create one
-    if user_id.to_i == 0
-      service_call = create_members(send_notification: false)
-      user_id = service_call.result.user_id
+    overall_result = nil
+
+    find_or_create_users(send_notification: false) do |member_params|
+      service_call = WorkPackageMembers::CreateOrUpdateService
+                      .new(user: current_user)
+                      .call(entity: @work_package,
+                            user_id: member_params[:user_id],
+                            role_ids: find_role_ids(params[:member][:role_id]))
+
+      @share = service_call.result
+
+      if overall_result
+        overall_result.push(service_call)
+      else
+        overall_result = [service_call]
+      end
     end
 
-    @share = WorkPackageMembers::CreateOrUpdateService
-      .new(user: current_user)
-      .call(entity: @work_package,
-            user_id:,
-            role_ids: find_role_ids(params[:member][:role_id])).result
-
+    @shares = overall_result.map(&:result)
 
-    if current_visible_member_count > 1
-      respond_with_prepend_share
-    else
-      respond_with_replace_modal
+    unless overall_result.nil?
+      if current_visible_member_count > 1
+        respond_with_prepend_share
```

Comment:
Let's rename this to

```suggestion
        respond_with_prepend_shares
```